director/template/var_kv_test.go

```diff
@@ -23,6 +23,25 @@ var _ = Describe("VarKV", func() {
 			Expect(arg).To(Equal(VarKV{Name: "name", Value: "val"}))
 		})
 
+			It("reverts to the old (incorrect) bevaviour and removes the wrapping double quotes if the value is a string", func() {
+				err := (&arg).UnmarshalFlag(`name="val"`)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(arg).To(Equal(VarKV{Name: "name", Value: "val"}))
+			})
+			It("reverts to the old (incorrect) bevaviour and removes the wrapping single quotes if the value is a string", func() {
+				err := (&arg).UnmarshalFlag(`name='val'`)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(arg).To(Equal(VarKV{Name: "name", Value: "val"}))
+			})
+			It("Trim only removes wrapping quotes", func() {
```

Comment:
For completeness's sake, can we add in a test or three here that ensures that values with unbalanced quote characters are correctly handled? For example
`"'some-data"'` or `'"some-data'"` get transformed into `'some-data"` and `"some-data'`, respectively?

I expect this to work _today_, but it'd be good to ensure that it keeps working in the future.

Reply:
There was a syntax issue with the test which I fixed. I wanted to add the suggested cases here but they don't work at the moment. You get errors like:
```
  Unexpected error:
        <errors.ComplexError>: {
            Err: <*errors.errorString | 0xc00039ac80>{
                s: "Deserializing variables 'name='\"some-data'\"'",
            },
            Cause: <*errors.errorString | 0xc00039ac50>{
                s: "yaml: found unexpected end of stream",
            },
        }
        Deserializing variables 'name='"some-data'"': yaml: found unexpected end of stream
    occurred
```
Is this expected to work?

Reply:
I think it should since that could be a Passwort Value:

`"'someString'"` should be untouched  and render to `"'someString'"` 
`"'someString"` should be touched  and render to `'someString`